src/app/v2/trade/[primary]/[numeraire]/page.tsx

```diff
@@ -0,0 +1,80 @@
+'use client';
+
+import { PairSelector } from '@/components/PairSelector';
+import { Chart } from '@/components/chart';
+import { RouteBook } from '@/components/route-book';
+import { Card } from '@penumbra-zone/ui/Card';
+import { useAssets } from '@/shared/state/assets.ts';
+import { useMemo } from 'react';
+
+interface QueryParams {
+  primary: string;
+  numeraire: string;
+}
+
+// Converts symbol to Metadata
+const usePathToMetadata = (params: QueryParams) => {
+  const { data, error, isLoading } = useAssets();
+  return useMemo(
+    () => ({
+      primary: data?.find(a => a.symbol === params.primary),
+      numeraire: data?.find(a => a.symbol === params.numeraire),
+      error,
+      isLoading,
+    }),
+    [data, error, isLoading, params.numeraire, params.primary],
+  );
+};
+
+export default function TradePage({ params }: { params: QueryParams }) {
+  const { primary, numeraire, error, isLoading } = usePathToMetadata(params);
+
+  return (
+    <div>
+      <div className='flex gap-2'>
+        {!!error && `Error loading pair selector: ${String(error)}`}
+        {isLoading && 'Loading...'}
+        {primary && numeraire && <PairSelector primary={primary} numeraire={numeraire} />}
+      </div>
+
+      <div className='flex flex-wrap lg:gap-2'>
+        <div className='w-full lg:w-auto lg:flex-grow mb-2'>
+          <Card title='Chart'>
+            {!!error && `Error loading chart: ${String(error)}`}
+            {isLoading && 'Loading...'}
+            {primary && numeraire && <Chart height={512} primary={primary} numeraire={numeraire} />}
+          </Card>
+        </div>
+        <div className='w-full sm:w-1/2 sm:pr-1 lg:w-[336px] lg:pr-0 mb-2'>
+          <Card title='Route Book'>
+            {!!error && `Error loading route book: ${String(error)}`}
+            {isLoading && 'Loading...'}
+            {primary && numeraire && <RouteBook primary={primary} numeraire={numeraire} />}
```

Comment:
We will have to build loading skeletons soon. Rough work in that direction. But mainly these components no longer take hard coded values and display data from the assets in the url.

Reply:
I believe here the RouteBook and Chart, shouldn't accept primary and numeraire as props, but re-use the usePathToMetadata (using next-router within that hook)